### PR TITLE
Add Go Client to Test Negative HTTP2 Conditions

### PR DESCRIPTION
**Pull request by ncteisen**
These tests are for the new Http/2 test server (https://github.com/grpc/grpc/pull/8900/). They are designed to test client behavior when the server sends unexpected responses during rpcs.

cc @ericgribkoff @makdharma

### CODE REVIEW

File: interop/http2/negative_http2_client.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		grpclog.Fatal("/TestService/UnaryCall RPC failed: ", err)
+	}
+	t := reply.GetPayload().GetType()
+	s := len(reply.GetPayload().GetBody())
+	if t != testpb.PayloadType_COMPRESSABLE || s != smallReqSize {
+		grpclog.Fatalf("Got the reply with type %d len %d; want %d, %d", t, s, testpb.PayloadType_COMPRESSABLE, smallReqSize)
+	}
+}
+
+func goaway(tc testpb.TestServiceClient) {
```
Comment (menghanl) [inline interop/http2/negative_http2_client.go]: How will this work? How to control the server behavior?

Comment (ncteisen) [inline interop/http2/negative_http2_client.go]: The validation for this test all occurs on the server side. The server sends a GOAWAY after the first rpc, and then asserts that the next rpc comes in on a different stream.

**Comment by ncteisen** [comment]
Addressed all comments. Also the official documentation for this is WIP, and will be added to grpc/grpc/docs by EOD

**Comment by ncteisen** [comment]
Documentation for the new tests [here](https://github.com/grpc/grpc/blob/master/doc/negative-http2-interop-test-descriptions.md)

File: interop/http2/negative_http2_client.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+func smallSimpleRequest() *testpb.SimpleRequest {
+	pl := clientNewPayload(testpb.PayloadType_COMPRESSABLE, smallReqSize)
+	return &testpb.SimpleRequest{
+		ResponseType: testpb.PayloadType_COMPRESSABLE.Enum(),
+		ResponseSize: proto.Int32(int32(smallReqSize)),
+		Payload:      pl,
+	}
+}
+
+func doSmallUnaryCall(tc testpb.TestServiceClient) {
```
Comment (menghanl) [inline interop/http2/negative_http2_client.go]: In the documentation, the request size and the response size are the same with the sizes used in `interop` package (different from the number you are using here though).
Can you reuse the function `DoLargeUnaryCall` defined in `interop` instead of defining a new function here?

**Comment by ncteisen** [comment]
Done

**Comment by ncteisen** [comment]
retest this please (does travis speak the jenkins language??)

**Comment by ncteisen** [comment]
Travis is happy, this should be good to go

**Comment by ncteisen** [comment]
I screwed this up trying to rebase. Fixing now

**Comment by ncteisen** [comment]
So to address the comments we spoke about last week, I added a sleep in the GOAWAY test. It's not the cleanest solution, but I think it should work well enough for now.

I put in an issue, #9300, to track implementing the "wait for GOAWAY" behavior in all of the client languages.

**Comment by ncteisen** [comment]
@makdharma and I intend to get finish up these negative http2 interop tests by the end of the week. Are the changes I made sufficient? Though there is a race condition for the goaway test, this behavior is consistent for all client languages.

**Comment by ncteisen** [comment]
I see @menghanl is OOO for a few weeks. Adding @MakMukhi.

**Comment by ncteisen** [comment]
Ping
